components/sup-protocol/src/lib.rs

```diff
@@ -69,7 +71,8 @@ use core::env as henv;
 use net::{ErrCode, NetResult};
 use rand::RngCore;
 use std::fs::File;
-use std::io::Read;
+use std::io::prelude::*;
+use std::io::BufReader;
```

Comment:
I think the fix can be made with a simpler change that obviates these imports. See below.

Reply:
agreed!